data/lib/service/team/team_service.dart

```diff
@@ -27,99 +27,81 @@ class TeamService {
+        return player.copyWith(user: user);
+      }).toList();
+
+      return team?.copyWith(players: players ?? []) ??
+          deActiveDummyTeamModel(teamId);
     } catch (error, stack) {
       throw AppError.fromError(error, stack);
     }
   }
 
-  Stream<TeamModel> getTeamStreamById(String teamId) {
+  Stream<TeamModel> streamTeamById(String teamId) {
     return _teamsCollection.doc(teamId).snapshots().asyncMap((teamDoc) async {
-      final teamRequestModel = teamDoc.data();
-      if (teamRequestModel == null) {
-        return deActiveDummyTeamModel(teamId);
-      }
-      final member = (teamRequestModel.players?.isNotEmpty ?? false)
-          ? await getMemberListFromUserIds(teamRequestModel.players ?? [])
-          : null;
-
-      final team = TeamModel(
-          name: teamRequestModel.name,
-          name_lowercase: teamRequestModel.name_lowercase,
-          id: teamRequestModel.id,
-          city: teamRequestModel.city,
-          created_at: teamRequestModel.created_at,
-          created_by: teamRequestModel.created_by,
-          profile_img_url: teamRequestModel.profile_img_url,
-          players: member);
-      return team;
+      final team = teamDoc.data();
+
+      final users = await getMemberListFromUserIds(
+        team?.players.map((e) => e.id).toList() ?? [],
+      );
+
+      final players = team?.players.map((player) {
+        final user = users.firstWhere((element) => element.id == player.id);
+        return player.copyWith(user: user);
+      }).toList();
+
+      return team?.copyWith(players: players ?? []) ??
+          deActiveDummyTeamModel(teamId);
     }).handleError((error, stack) => AppError.fromError(error, stack));
   }
 
   Stream<List<TeamModel>> getUserRelatedTeams() {
     if (_currentUserId == null) {
       return Stream.value([]);
     }
-    final filter = Filter.or(
-      Filter(FireStoreConst.createdBy, isEqualTo: _currentUserId),
-      Filter(FireStoreConst.players, arrayContains: _currentUserId),
-    );
-    return _teamsCollection
-        .where(filter)
-        .snapshots()
-        .asyncMap((snapshot) async {
-      return await Future.wait(snapshot.docs.map((mainDoc) async {
-        AddTeamRequestModel teamRequestModel = mainDoc.data();
-
-        final member = (teamRequestModel.players?.isNotEmpty ?? false)
-            ? await getMemberListFromUserIds(teamRequestModel.players ?? [])
-            : null;
-
-        final team = TeamModel(
-          name: teamRequestModel.name,
-          name_lowercase: teamRequestModel.name_lowercase,
-          id: teamRequestModel.id,
-          city: teamRequestModel.city,
-          created_at: teamRequestModel.created_at,
-          created_by: teamRequestModel.created_by,
-          profile_img_url: teamRequestModel.profile_img_url,
-          players: member,
-        );
-        return team;
-      }));
+    return _teamsCollection.snapshots().asyncMap((snapshot) async {
```

Comment:
we're fetching all data and then sorting from it, it's not right way to do so, we can use data duplication for this, how about keeping players field as it is and adding another field called admins and it includes id of admins?